arrays/c-or-cpp/unique-elements-in-an-array.c

```diff
@@ -0,0 +1,32 @@
+#include<stdio.h>
+
+int main(){
+    int a[7];
+    
+    for(int k=0; k<7; k++){
+        scanf("%d", &a[k]);
```

Comment:
prefer to either ask user for size before, or create a default array with the some sample values to test this algo on.

Reply:
Does this code properly deal with negative numbers?

Reply:
@cclauss yes it does.

Reply:
`"Enter size of the array: " -5`

Reply:
> @cclauss yes it does.

@Arsenic-ATG according to the code written it does take negative number as size if he wanted to do that he can check that the given size is negative if yes size=size*-1;

Reply:
@cclauss oh, I though you were talking about negative numbers as array elements.

Well the main focus of this program( and I think all the other programs here ) is implementation of algorithm ( finding unique elements in an array ), considering the inputs are given correctly and in reasonable range, not able to handle negative size is actually not the algorithm's fault. I don't expect people who are visiting this repo to use these programs as they are, they are mainly there for educational and learning purposes, so that anyone can view the program and learn about how the algorithm is being implemented in a particular language, and adding additional checks to check for bounds of algo can make it overwhelming for newcomers especially when the algorithm is implemented directly in main() and not as a different function.

But yes, I do agree that it's a good practice to keep it free from all human errors also and adding those additional checks will always be appreciated.

Reply:
I know that this is already merged but please allow me to make a few comments...

> when the algorithm is implemented directly in main()

I believe that is an anti-pattern because such an algorithm is very difficult to move into a larger program and is complicated to test.  I think my contributions over the past week have demonstrated the value of testing algorithms.  Without tests, even simple algorithms deliver results that are different from our intuition.  (See the comments on my PRs.). Algorithms that come with tests are more dependable and can be modified/improved with confidence.  Writing tests help us to find problems before our users do.

> considering the inputs are given correctly and in reasonable range

This is called [“happy path”](https://en.wikipedia.org/wiki/Happy_path) programming and it is a really poor pattern to teach to newcomers.  We should encourage contributors (even first-time contributors) to be on the lookout for corner cases that will make software fragile in the real world.  We do not our contributors to fail future job interviews because they only know how to code for the happy path.

Reply:
Good point

Reply:
Ok @cclauss, I will be stricter while reviewing the pull requests from now on 🙂👍

and as for this pull request is concerned, I will be updating the program soon to meet the requirements.